examples/zephyr/http-client/Makefile

```diff
@@ -1,59 +1,40 @@
-MONGOOSE_ROOT ?= $(realpath $(CURDIR)/../../..)
-DOCKER_PROJECT_DIR ?= /workdir
-ZEPHYR_DIR ?= zephyrproject
-PROJECT_NAME = $(notdir $(CURDIR))
-PROJECT_PATH = $(realpath $(CURDIR))
-ZEPHYR_PATH = $(realpath $(CURDIR)/..)/$(ZEPHYR_DIR)
-DOCKER_PROJECT_PATH = $(DOCKER_PROJECT_DIR)/$(PROJECT_NAME)
-DOCKER_ZEPHYR_PATH = $(DOCKER_PROJECT_DIR)/$(ZEPHYR_DIR)
-
-BOARD ?= qemu_x86
-ifeq "$(BOARD)" "qemu_x86"
-OVERLAY ?= -DOVERLAY_CONFIG=overlay-e1000.conf
-else
-OVERLAY ?= 
-endif
-
-SHAREUSB ?= --privileged -v /dev/bus/usb/:/dev/bus/usb
-SHARETUN ?= --cap-add=NET_ADMIN --device /dev/net/tun --net=host
-DOCKER ?= docker run --rm -v $(PROJECT_PATH):$(DOCKER_PROJECT_PATH) -v $(ZEPHYR_PATH):$(DOCKER_ZEPHYR_PATH)
-REPO ?= zephyrprojectrtos/ci
-
-
-example:
+TOP_DIR ?= $(realpath $(CURDIR)/../../..)
+ZEPHYR_DIR ?= $(TOP_DIR)/zephyrproject
+DOCKER ?= docker run --rm \
+          -v $(TOP_DIR):$(TOP_DIR) \
+          -w $(realpath $(CURDIR)) \
+          -e ZEPHYR_BASE=$(ZEPHYR_DIR)/zephyr \
+          $(DOCKER_ARGS) zephyrprojectrtos/ci
+#BUILD_ARGS ?= -b qemu_x86 -DOVERLAY_CONFIG=overlay-e1000.conf
+BUILD_ARGS ?= -b nucleo_f429zi
+
+.PHONY: build flash clean pristine run
+
+example zephyr:
 	true
 
-build:
-	cp $(MONGOOSE_ROOT)/mongoose.[c,h] src/
-	$(DOCKER) $(REPO) /bin/sh -c 'cd $(DOCKER_ZEPHYR_PATH)/zephyr && \
-	west build -b $(BOARD) -p auto $(DOCKER_PROJECT_PATH) $(OVERLAY) --build-dir $(DOCKER_PROJECT_PATH)/build'
+$(ZEPHYR_DIR):
+	mkdir -p $@
+	$(DOCKER) west init $@
+	$(DOCKER) sh -c 'cd $@ && west update'
 
+build: $(ZEPHYR_DIR)
+	cp $(TOP_DIR)/mongoose.[c,h] src/
+	$(DOCKER) west build $(BUILD_ARGS) -p auto $(realpath $(CURDIR))
+
+run: DOCKER_ARGS ?= --cap-add=NET_ADMIN --device /dev/net/tun --net=host -it
 run:
-	$(DOCKER) -it $(SHARETUN) $(REPO) /bin/sh -c 'cd $(DOCKER_ZEPHYR_PATH) && \
-	west build -t run --build-dir $(DOCKER_PROJECT_PATH)/build'
+	$(DOCKER) west build -t run
 
 clean:
-	$(DOCKER) $(REPO) /bin/sh -c 'cd $(DOCKER_ZEPHYR_PATH) && \
-	west build -t clean --build-dir $(DOCKER_PROJECT_PATH)/build'
+	rm -rf $(TOP_DIR)/.west $(TOP_DIR)/zephyrproject build
```

Comment:
They have their own partial cleaning (_clean_, keeping board config) and full cleaning (_pristine_, erasing everything)
If you go `rm` that doesn't work on some systems (like mine...) where Docker runs as root, as you don't own those Docker generated files and need to sudo to rm them. The clean operation through Docker is to keep everything the Zephyr way, so Zephyr developers are happy. There is a noticeable speedup when you clean and recompile vs when you pristine and recompile

Reply:
Yeah I overlooked this... Added west pristine and west clean. I think since docker is run with ZEPHYR_BASE environment set, we're good and don't need to `cd ...`